Salehi2025/Figure5.qmd

```diff
@@ -0,0 +1,450 @@
+agg_counts <- counts_df %>%
+  group_by(sample, Class) %>%
+  summarize(class_total = sum(count, na.rm = TRUE), .groups = "drop")
+
+# Keep only the top 5 most frequent antibiotic classes
+top_5_AB_classes <- c(
+  "Tetracycline", 
+  "Beta-lactam", 
+  "Macrolide, Lincosamide, Streptogramin B", 
+  "Aminoglycoside", 
+  "Amphenicol"
+)
+
+agg_counts <- agg_counts %>% filter(Class %in% top_5_AB_classes)
+```
+
+This section prepares AB class count data for modeling by: 1. Adding a small pseudocount to avoid log(0) 2. Converting long format to wide (samples x AB classes) 3. Merging with metadata 4. Applying log transformation to the top 5 AB classes
+
+```{r, pseudocount}
+# Add pseudocount to avoid log(0)
+agg_counts$class_total <- ifelse(agg_counts$class_total == 0,
+                                 min(agg_counts$class_total[agg_counts$class_total != 0]),
+                                 agg_counts$class_total)
+
+# Reshape data: one row per sample, columns = AB classes
+agg_counts_wide <- agg_counts %>%
+  spread(key = "Class", value = "class_total")
+
+# Filter only samples that exist in metadata
+agg_counts_wide <- agg_counts_wide %>%
+  filter(sample %in% adult_metadata$acc)
+
+# Merge wide-format AB class counts with metadata
+adult_metadata <- full_join(adult_metadata %>% mutate(sample = acc),
+                            agg_counts_wide, 
+                            by = "sample")
+
+# Log-transform the top 5 antibiotic class counts
+top5_log <- adult_metadata[, top_5_AB_classes] %>% log
+
+# Rename columns to indicate log-transformed values
+colnames(top5_log) <- paste0("log_", gsub(" ", "_", colnames(top5_log)))
+
+# Add log-transformed AB class values to metadata
+adult_metadata <- cbind(adult_metadata, top5_log)
+```
+
+This section prepares the data for beta diversity analysis and computes Shannon diversity index from relative abundance profiles.
+
+```{r, Beta Diversity}
+# Filter metadata to include only samples with GDP and antibiotic usage information
+adult_metadata <- adult_metadata %>%
+  mutate(
+    continent = factor(geo_loc_name_country_continent_calc)
+  ) %>%
+  filter(!is.na(Usage), !is.na(GDP_per_head))
+
+# Keep only samples with non-missing GDP and antibiotic usage in the TSE object
+non_na_conditions <- !is.na(colData(TSE_filtered)$GDP_per_head) &
+                     !is.na(colData(TSE_filtered)$Usage)
+TSE_adult_filtered <- TSE_filtered[, non_na_conditions]
+
+# Update the sample metadata in the filtered TSE object
+colData(TSE_adult_filtered) <- DataFrame(adult_metadata)
+
+# Extract relative abundance data
+assay_data_clean <- assay(TSE_adult_filtered, "relabundance")
+
+# Remove samples that contain only zero abundances
+non_empty_samples <- colSums(assay_data_clean > 0, na.rm = TRUE) > 0
+assay_data_clean <- assay_data_clean[, non_empty_samples]
+
+# Subset metadata to match samples with non-zero data
+adult_metadata <- colData(TSE_adult_filtered)[non_empty_samples, ]

```

Comment:
move all data preparation into separate script and see if there is something that can be shortedned / removed / cleaned up